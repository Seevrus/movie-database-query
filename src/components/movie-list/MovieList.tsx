import { Container, Row } from 'react-bootstrap';
import { Movie } from '../../model/Movie';
import { MovieCard } from './MovieCard';

interface MovieListProps {
  movies: Movie[];
}

export const MovieList = ({ movies }: MovieListProps) => {
  return (
    <Container fluid>
      <Row className="justify-content-lg-center">
        {movies.map((movie) => <MovieCard key={movie.id} movie={movie} />)}
      </Row>
    </Container>
  );
};
