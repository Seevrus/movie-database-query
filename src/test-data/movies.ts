import { Movie } from '../model/Movie';

const movies = [
  new Movie({
    id: 1,
    title: 'Reality Bites',
  }),
  new Movie({
    id: 2,
    title: 'Demolition Man',
  }),
  new Movie({
    id: 3,
    title: 'Kramer vs. Kramer',
  }),
  new Movie({
    id: 4,
    title: 'The Christmas Chronicles: Part Two',
  }),
  new Movie({
    id: 5,
    title: 'The Ides of March',
  }),
  new Movie({
    id: 6,
    title: 'The Princess Switch 3: Romancing the Star',
  }),
];

export default movies;
