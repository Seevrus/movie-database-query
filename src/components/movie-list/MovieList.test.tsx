import renderer from 'react-test-renderer';
import movies from '../../test-data/movies';
import { MovieList } from './MovieList';

describe('MovieList', () => {
  it('should render correctly if there are no movies', () => {
    const component = renderer.create(
      <MovieList isLoading={false} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correctly if results are loading', () => {
    const component = renderer.create(
      <MovieList isLoading movies={[]} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const component = renderer.create(
      <MovieList isLoading={false} movies={movies} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
