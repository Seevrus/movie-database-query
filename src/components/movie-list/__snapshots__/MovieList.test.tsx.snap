// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-lg-center row"
  >
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of Reality Bites"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Reality Bites
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of Demolition Man"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Demolition Man
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of Kramer vs. Kramer"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Kramer vs. Kramer
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of The Christmas Chronicles: Part Two"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              The Christmas Chronicles: Part Two
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of The Ides of March"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              The Ides of March
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "maxWidth": "1280px",
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-3"
        >
          <img
            alt="Poster of The Princess Switch 3: Romancing the Star"
            className="img-fluid"
            src=""
          />
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              The Princess Switch 3: Romancing the Star
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              />
            </p>
            <button
              className="btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Kedvencek közé
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieList should render correctly if results are loading 1`] = `
<div
  className="d-flex justify-content-center spinner-container container"
>
  <div
    className="spinner-border text-secondary"
    role="status"
  />
</div>
`;

exports[`MovieList should render correctly if there are no movies 1`] = `null`;
